rust/db_handling/Cargo.toml

```diff
@@ -1,30 +1,34 @@
 [package]
 name = "db_handling"
 version = "0.1.0"
-authors = ["vera"]
-edition = "2018"
+edition = "2021"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-sled = "0.34.6"
-parity-scale-codec = "2.2.0"
-parity-scale-codec-derive = "2.2.0"
-hex = "0.4.3"
-regex = "1.4.5"
-lazy_static = "1.4.0"
-sp-core = "3.0.0"
-sp-runtime = "3.0.0"
-tiny-bip39 = { version = "0.8.0", default-features = false }
-blake2-rfc = "0.2.18"
-zeroize = "1.5.2"
-constants = {path = "../constants"}
-defaults = {path = "../defaults"}
-definitions = {path = "../definitions"}
-qrcode_static = {path = "../qrcode_static"}
+blake2-rfc = {version = "0.2.18", optional = true}
 chrono = "0.4.19"
-plot_icon = "0.1.3"
-png = "0.16.8"
+constants = {path = "../constants"}
+defaults = {path = "../defaults", default-features = false, optional = true}
+definitions = {path = "../definitions", default-features = false}
+hex = {version = "0.4.3", optional = true}
+lazy_static = "1.4.0"
+parity-scale-codec = {version = "3.1.2", features = ["derive"]}
+plot_icon = {version = "0.2.0", default-features = false, features = ["pix"], optional = true}
+qrcode_static = {path = "../qrcode_static", optional = true}
+regex = "1.5.4"
+sled = "0.34.6"
+sp-core = {git = "https://github.com/paritytech/substrate", default-features = false, features = ["full_crypto"], optional = true}
+sp-runtime = {git = "https://github.com/paritytech/substrate", default-features = false, optional = true}
+tiny-bip39 = {version = "0.8.0", default-features = false, optional = true}
+zeroize = {version = "1.5.3", optional = true}
+
+[features]
+default = ["test"]
+
+active = ["constants/active", "defaults/active", "definitions/active", "sp-core", "sp-runtime", "zeroize"]
+signer = ["blake2-rfc", "constants/signer", "defaults/signer", "definitions/signer", "hex", "plot_icon", "qrcode_static", "tiny-bip39", "sp-core", "sp-runtime", "zeroize"]
+test = ["active", "constants/test", "signer"]
```

Comment:
It would be better to break into multiple lines with one feature per line, see [this](https://github.com/paritytech/substrate/blob/master/frame/grandpa/Cargo.toml#L46) for example